RakutenShop/Scenes/RecipeDetailView/RecipeDetailViewController.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  RecipeDetailViewController.swift
+//  RakutenShop
+//
+//  Created by Garnik Ghazaryan on 2/11/20.
+//  Copyright © 2020 Garnik Ghazaryan. All rights reserved.
+//
+
+import UIKit
+import ReactiveCocoa
+import ReactiveSwift
+import AlamofireImage
+import SVProgressHUD
+
+class RecipeDetailViewController: BaseViewController {
+
+    @IBOutlet weak var recipeImageView: UIImageView!
+    @IBOutlet weak var instructionsTableView: UITableView!
+    @IBOutlet weak var similarRecipesCollectionView: UICollectionView!
+    @IBOutlet weak var caloriesLabel: UILabel!
+    @IBOutlet weak var carbsLabel: UILabel!
+    @IBOutlet weak var fatsLabel: UILabel!
+    @IBOutlet weak var proteinsLabel: UILabel!
+    
+    var viewModel: RecipeDetailViewViewModel?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        instructionsTableView.dataSource = self
+        instructionsTableView.delegate = self
+        instructionsTableView.tableFooterView = UIView()
+        
+        similarRecipesCollectionView.dataSource = self
+        similarRecipesCollectionView.delegate = self
+        
+        // Do any additional setup after loading the view.
+    }
+    
+    override func setupView() {
+        let rightBarItem = UIBarButtonItem(title: "Save", style: .plain, target: self, action: #selector(saveButtonTapped))
+        self.navigationItem.setRightBarButton(rightBarItem, animated: true)
+    }
+    
+    override func setupViewModel() {
+        guard let viewModel = viewModel else { return }
+        
+        viewModel.isLoading.producer.startWithValues { (showLoading) in
+            DispatchQueue.main.async {
+                showLoading ? SVProgressHUD.show() : SVProgressHUD.dismiss()
+            }
+        }        
+        
+        viewModel.errorMessage.producer.skip(first: 1).startWithValues { (errorMessage) in
+            DispatchQueue.main.async {
+                SVProgressHUD.showError(withStatus: errorMessage)
+            }
+        }
+        
+        viewModel.recipeObject?.producer.startWithValues({ [unowned self] (repiceObject) in
+            DispatchQueue.main.async {
+                self.updateUI(recipeObject: repiceObject)
+            }
+        })
+        
+        viewModel.nutritionsObject.producer.startWithValues({ [unowned self] (nutritionsObject) in
+            DispatchQueue.main.async {
+                self.updateNutritions(nutritionsObject: nutritionsObject)
+            }
+        })
+        
+//        viewModel.analyzedInstructions.producer.startWithValues({ (instructions) in
+//            print(instructions)
+//        })
+    
+        reactive.makeBindingTarget { (weakself, object) in
+            self.instructionsTableView.reloadData()
```

Comment:
self -> weakself
instead of ojbect -> _ just underscore

the same for next one.